src/middlewares/validate.ts

```diff
@@ -0,0 +1,38 @@
+import { schema } from "./users.joiValidation";
+import { Request, Response, NextFunction } from "express";
+import { NOT_ACCEPTABLE } from "../constants/statusCode";
+
+export const validation = (User: { rawAttributes: {} }) => {
```

Comment:
Please rename it to user validation as still its not generic validation. If you want to make it generic validation then you would get the columns/fields from the given schema and iterate over its properties.